neuralmonkey/dataset.py

```diff
@@ -77,28 +91,49 @@ def _get_name_from_paths(series_paths: Dict[str, str]) -> str:
     Returns:
         The name for the dataset.
     """
+
     name = "dataset"
-    for _, path in series_paths.items():
-        name += "-{}".format(path)
+    for paths, _ in series_paths.values():
+        name += "-{}".format("+".join(paths))
     return name
 
 
-def _get_series_paths(kwargs: Dict[str, str]) -> Dict[str, str]:
-    """Get paths to files that contain data from the dataset keyword
-    argument specs.
+def _get_series_paths_and_readers(
+        kwargs: Dict[str, Any]) -> Dict[str, Tuple[List[str], Reader]]:
+    """Get paths to files that contain data from the dataset kwargs.
 
-    Input file for a serie named 'xxx' is specified by parameter 's_xxx'
+    Input file for a serie named 'xxx' is specified by parameter 's_xxx'. The
+    dataset series is define by a string with a path / list of strings with
+    path, or a tuple whose first member is path or list of path and the second
+    memeber is a reader function.
```

Comment:
Mimochodem, co to teda znamená, že tam můžu dát buď string a to bude cesta, anebo tuple, a první bude cesta nebo seznam cest a druhý bude reader?